### ISSUE TITLE
Export All TypeScript Types

### ISSUE THREAD

**Issue by cheetamaru** [lionralfs/discogs-client#295]
Hello,

Firstly, I'd like to say thank you for this library. It's been incredibly useful in my project.

## Issue:
I've noticed that not all TypeScript types are exported from the library, which makes it difficult to use certain features in a type-safe manner.

For example:

`GetProfileResponse` from `https://github.com/lionralfs/discogs-client/blob/main/lib/user.ts`
`WantlistEntryResponse` from `https://github.com/lionralfs/discogs-client/blob/main/lib/wantlist.ts`

## Request:
Could you please export all the TypeScript types from the library? This includes types, type aliases, unions and any other types defined within the codebase and based on Discogs API.

## Benefits:

- Enhanced Developer Experience: Developers can more easily understand and use the library when all types are accessible, providing better IDE support and autocompletion.
- Consistent Codebase: Allows for more consistent use of the library across different projects and teams.

## Workaround:
Currently, to access the necessary types, I have to use `Awaited<ReturnType<typeof T>>` of my custom `T` method.

```ts
import apiClient from '@/service/api/apiClient'

export type DiscogsUser = Awaited<ReturnType<typeof getUser>>;
 
export const getUser = async (username: string) => {
    const user = await apiClient.user().getProfile(username);

    return user.data
}

```
---
Thank you for considering this request. I believe it will make the library even more robust and user-friendly for the community.

**Comment by lionralfs** [comment]
Hey, amazing (and well written) feature request ❤️

I've just merged #296 and released the changes as version `4.1.0`. Let me know if I've missed anything

**Comment by cheetamaru** [comment]
Wow, that was quick!! Really appreciate it. Thanks a lot!
